plugins/node/opentelemetry-instrumentation-ioredis/src/internal-types.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import type { Command, Redis } from 'ioredis';
+import type * as LegacyIORedis from 'ioredis4';
```

Comment:
This still depends on ioredis. It will fail to compile if ioredis is missing from node_modules. You will have to vendor the types

Reply:
`internal-types.ts` is not exported from `index.d.ts` so it should be ok I think